src/app/modules/main/profile_section/devices/view.nim

```diff
@@ -133,17 +129,15 @@ QtObject:
 
   # LocalPairing actions
 
-  proc userAuthenticated*(self: View, pin: string, password: string, keyUid: string) {.signal.}
-  proc emitUserAuthenticated*(self: View, pin: string, password: string, keyUid: string) =
-    self.userAuthenticated(pin, password, keyUid)
-  proc authenticateUser*(self: View, keyUid: string) {.slot.} =
-    self.delegate.authenticateUser(keyUid)
+  proc openPopupWithConnectionStringSignal*(self: View, rawConnectionString: string) {.signal.}
```

Comment:
> Why change userAuthenticated to openPopupWithConnectionStringSignal and hide the logic for getConnectionString... inside?

`userAuthenticated` signal is not needed at all in this approach, it's not changed with `openPopupWithConnectionStringSignal`, but removed.

> Now authenticateUser will automatically call getConnectionStringForBootstrappingAnotherDevice, which doesn't make sense to me.

How did you conclude that `authenticateUser` automatically call `getConnectionStringForBootstrappingAnotherDevice`? 
Firstly that is not true, please take a closer look and you will see.

Secondly what's the purpose of authenticating a user?

> Previously the logic was quite obvious

Depends how you look at it 😄, I see this like an improved approach and will explain now why it's like that.

Before this was the flow:
1. qml initiate `setupSyncing`
2. qml ask backend to authenticate the user
3. backend authenticates the user running an authentication popup
4. when user provide credentials, backend verifies them 
5. backend sends authenticated details to qml
6. qml uses those details to ask backed to create a connection string based on those received details
7. backend creates a connection string and returns it to qml
8. qml displays connection string within SetupSyncingPopup

So to answer above asked question, the only reason of authentication here is to get authentication details like password in case of a regular user and public encrypted key in case of a keycard user, with the goal to use them to create a connection string.

Now this is the flow:
1. qml initiate `setupSyncing`
2. qml ask backend to authenticate the user
3. backend authenticates the user running an authentication popup
4. when user provide credentials, backend verifies them 
5. backend creates a connection string and sends it to qml
6. qml displays connection string within SetupSyncingPopup

So the flow is simpler now cause we don't need to unnecessarily bring it down to qml, which are points 5 and 6 from the old flow.

Reply:
> How did you conclude that authenticateUser automatically call getConnectionStringForBootstrappingAnotherDevice?
Firstly that is not true, please take a closer look and you will see.

hmmm.
But this is executed after successful authentication, i.e. is subsequent to `authenticateLoggedInUser`:
https://github.com/status-im/status-desktop/blob/7e65db4a3f8542c606c109f0592edde8881bba0e/src/app/modules/main/profile_section/devices/module.nim#L109-L117

That's why I say that it "automatically" leads to creation of connection string.

> So the flow is simpler now cause we don't need to unnecessarily bring it down to qml, which are points 5 and 6 from the old flow.

I see that this was your intention 🙂 

I personally feel confusing that calling a function `authenticateUser` does something much more that just what the naming says. In my opinion, although we do want a connection string at the end, we shouldn't hide this process behind `authenticateUser` call. So it's kinda `authenticateAndGetConnectionString...` function now. But I just don't like those `...And...` functions, to me it's more readable to separate them.

Probably the other way around make more sense to me, i.e. hide authentication behind the `getConnectionString...`, because we indeed need a password for that procedure, so we automatically open auth popup.

Also, imagine that we need auth for some other procedure as well, e.g. enable syncing with non-paired device (not a real feature yet, but potentially possible). We would have to authenticate with same `devices` module, but need another backend after all.

This is all just my opinion, I don't insist to leave the previous solution. 
It's up to you 🙂

Reply:
Well it's a question whether qml drives the logic or backend.

I guess that you won't be confused if I named `authenticateLoggedInUser` function `resolveConnectionString` which will run authentication popup and generate connection string on successful authentication and send it to qml.

Reply:
> Also, imagine that we need auth for some other procedure as well

Will rename the method to avoid confusion, no worries ;)

> e.g. enable syncing with non-paired device (not a real feature yet, but potentially possible)

this sounds a bit unreal ;) but I see the point, will rename the method